### PR TITLE
Major code cleanup

### PR DESCRIPTION
**Pull request by Nyholm**
#### Breaking changes: 
* Made all properties private -> use functions to access properties of our objects
* Removed duplicated class `Oauth2Client` -> Use `Auth0`
* Fixed typo in `persistantMap` -> Renamed to `persistentMap`

#### Code style
* Removed empty file `ApiClient`
* Improved doc blocks
* Use camelCase instead of snake_case
* Fixed indentation

Sorry. I know this PR is horrible to review.

### CODE REVIEW

**Comment by cocojoe** [comment]
If there are public API changes please add the `before -> after` to first comment.  Thx

**Comment by Nyholm** [comment]
Im done with this PR now. I will create a new PR for cleaning up `Authentication`. That class will include some more BC breaks.

File: src/API/Authentication.php [hunk 1]
```diff
@@ -150,198 +159,199 @@ public function email_passwordless_start($email, $type, $authParams = [])
...
+
+        return $this->oauth_token($options);
+    }
+
+    /**
+     * Makes a call to the `oauth/token` endpoint with `client_credentials` grant type.
+     *
+     * @method client_credentials
+     *
+     * @param {Object} options: https://auth0.com/docs/api-auth/grant/client-credentials
```
Comment (cocojoe) [inline src/API/Authentication.php]: Would rather have links to docs as @seealso or @notes or whatever the equivalent is in PHP Docs headers.  As it's not a param.

Comment (Nyholm) [inline src/API/Authentication.php]: I agree. I only changed indentation on this line. I'll improve this class in a next PR.

**Comment by glena** [comment]
@cocojoe LGTM

**Comment by Nyholm** [comment]
Awesome, Thank you for merging. 

A friendly reminder to enable StyleCI to make sure PRs comply with our coding standards. 

https://styleci.io/

**Comment by glena** [comment]
yeah, it is not that easy since it should be authorized by the security team in the auth0 org.

**Comment by Nyholm** [comment]
Ah, okey. Thanks

**Pull request merged**